backend.go

```diff
@@ -103,6 +108,11 @@ func Backend() *kubeAuthBackend {
 		),
 	}
 
+	// create and store a new DefaultPooledClient for connection reuse
+	if b.httpClient == nil {
+		b.httpClient = cleanhttp.DefaultPooledClient()
+	}
```

Comment:
`b.httpClient` will always be nil here, right? I wonder if this check would fit better in the Review function.

Reply:
Alternatively, maybe just set it directly in the `&framework.Backend{...}` struct initialisation above?

Reply:
@tvoran You are right! In commit [2d36b73](https://github.com/hashicorp/vault-plugin-auth-kubernetes/pull/142/commits/2d36b73054af33d72d407b211b39f7f6bf2768c4) I removed the nil check from the backend struct and added the nil check inside of service account where token review has access to the backend.

Reply:
@tomhjp I don't believe I need to set client inside of `&framework.Backend{...}` because the service lookup function already has access to the `kubAuthBackend` struct so all I had to do was pass the client as a reference to the only caller which is the Review function.

Reply:
Oh ignore me, I misread the code 👍 what you've got makes sense.